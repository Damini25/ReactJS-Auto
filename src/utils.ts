import axios from 'axios';

export const fetchCars=async (params:{})=>{
    const response = await axios({
        method: 'GET',
        url:'https://auto1-mock-server.herokuapp.com/api/cars',
        headers: {
            'Content-Type': 'application/json'
        },
        params: {...params}
    });
    return response.data
}

export const fetchColors=async (params:{})=>{
    const response = await axios({
        method: 'GET',
        url:'https://auto1-mock-server.herokuapp.com/api/colors',
        headers: {
            'Content-Type': 'application/json'
        },
        params: {...params}
    });
    return response.data
}

export const fetchManufacturers=async (params:{})=>{
    const response = await axios({
        method: 'GET',
        url:'https://auto1-mock-server.herokuapp.com/api/manufacturers',
        headers: {
            'Content-Type': 'application/json'
        },
        params: {...params}
    });
    return response.data
}

export const fetchCarDetails=async (stockNumber:any,params:{})=>{
    const response = await axios({
        method: 'GET',
        url:`https://auto1-mock-server.herokuapp.com/api/cars/${stockNumber}`,
        headers: {
            'Content-Type': 'application/json'
        },
        params: {...params}
    });
    return response.data
}

export const capitalize = (str: string) => {
    return str.charAt(0).toUpperCase() + str.slice(1);
}
export const getProp = (obj: any, prop = '', def = '') => {
    if (!obj) {
        return '';
    }
    if (!prop) {
        return obj;
    }
    return !obj[prop] ? def : obj[prop];
}
