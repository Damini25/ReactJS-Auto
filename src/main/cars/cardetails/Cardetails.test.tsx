import React from 'react';
import { render, act } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import CarDetails from './Cardetails';
import axios, { AxiosStatic} from 'axios';
import { fetchCarDetails } from '../../../utils';

interface AxiosMock extends AxiosStatic {
    mockResolvedValue: Function
    mockRejectedValue: Function
}

jest.mock('axios')
jest.mock('react-router-dom', () => {
    // Require the original module to not be mocked...
    const originalModule = jest.requireActual('react-router-dom');
    return {
        __esModule: true,
        ...originalModule,
        useParams: jest.fn(),
    };
});
const mockedAxios = axios as AxiosMock
const mockData = {
    "car": {
        "stockNumber": 10123,
        "manufacturerName": "Tesla",
        "modelName": "Model X",
        "color": "blue",
        "mileage": { "number": 153819, "unit": "km" },
        "fuelType": "Petrol",
        "pictureUrl": "https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"
    }
}
it('displays the cardetails from cardetails API', async () => {
    mockedAxios.mockResolvedValue({ data: mockData });
    await act(async () => {
        const res = await fetchCarDetails(10123, {})
        const { asFragment }=render(<Router><CarDetails /></Router>,)
        expect(res).toEqual(mockData)
        expect(asFragment()).toMatchSnapshot();
    })
})