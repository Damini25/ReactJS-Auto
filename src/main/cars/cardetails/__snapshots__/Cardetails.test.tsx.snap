// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the cardetails from cardetails API 1`] = `
<DocumentFragment>
  <div
    class="cardetails"
  >
    <div
      class="loading"
      data-testid="loading"
      id="loading"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
